### ISSUE TITLE
panic parsing `/********************************`

### ISSUE THREAD

**Issue by SergioBenitez** [raphlinus/pulldown-cmark#596]
The string

```
/********************************
```

causes the following panic:

```
thread 'main' panicked at 'attempt to shift right with overflow'
   in .cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.9.1/src/firstpass.rs:1836:21
stack backtrace:
   0: rust_begin_unwind
             at /rustc/76761db5919b6d2e178b26947ad9b4fcb0ff0e7c/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/76761db5919b6d2e178b26947ad9b4fcb0ff0e7c/library/core/src/panicking.rs:142:14
   2: core::panicking::panic
             at /rustc/76761db5919b6d2e178b26947ad9b4fcb0ff0e7c/library/core/src/panicking.rs:48:5
   3: pulldown_cmark::firstpass::simd::process_mask
             at .cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.9.1/src/firstpass.rs:1836:21
   4: pulldown_cmark::firstpass::simd::simd_iterate_special_bytes
             at .cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.9.1/src/firstpass.rs:1862:24
   5: pulldown_cmark::firstpass::simd::iterate_special_bytes
             at .cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.9.1/src/firstpass.rs:1811:22
   6: pulldown_cmark::firstpass::iterate_special_bytes
             at .cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.9.1/src/firstpass.rs:1556:9
   7: pulldown_cmark::firstpass::FirstPass::parse_line
             at .cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.9.1/src/firstpass.rs:460:13
   8: pulldown_cmark::firstpass::FirstPass::parse_paragraph
             at .cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.9.1/src/firstpass.rs:329:34
   9: pulldown_cmark::firstpass::FirstPass::parse_block
             at .cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.9.1/src/firstpass.rs:203:9
  10: pulldown_cmark::firstpass::FirstPass::run
             at .cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.9.1/src/firstpass.rs:53:18
  11: pulldown_cmark::firstpass::run_first_pass
             at .cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.9.1/src/firstpass.rs:35:5
  12: pulldown_cmark::parse::Parser::new_with_broken_link_callback
             at .cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.9.1/src/parse.rs:167:34
  13: pulldown_cmark::parse::Parser::new_ext
             at .cargo/registry/src/github.com-1ecc6299db9ec823/pulldown-cmark-0.9.1/src/parse.rs:154:9
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

Removing one asterisk `*` from the string resolves the panic.

The parser is invoked with:

```rust
let options = Options::all().difference(Options::ENABLE_SMART_PUNCTUATION);
let parser = Parser::new_ext(&input, options);
```

The panic occurs as a result of:

https://github.com/raphlinus/pulldown-cmark/blob/4d45c33e0ca450044dfb030b072a41978d709c98/src/firstpass.rs#L1815

This panic only occurs in debug mode. It occurs in `unsafe` code. It is thus possible that an invariant is violated in unsafe code that could lead to exploitation:

https://github.com/raphlinus/pulldown-cmark/blob/4d45c33e0ca450044dfb030b072a41978d709c98/src/firstpass.rs#L1800-L1821

**Comment by marcusklaas** [comment]
Oooh, that is an excellent find. I really thought that unsafe parts were watertight, but apparently not. I will investigate this more.

Thanks for the report!

**Comment by cheesycod** [comment]
can reprod

**Comment by Martin1887** [comment]
Fixed by #652.
